ada_feeding/ada_feeding/behaviors/blackboard_behavior.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Generic behavior with additional functionality for blackboard
+remappings (inspired by BehaviorTree.CPP)
+"""
+
+# Standard imports
+from typing import Any
+
+# Third-party imports
+import py_trees
+
+# Local imports
+from ada_feeding.helpers import BlackboardKey
+
+
+class BlackboardBehavior(py_trees.behaviour.Behaviour):
+    """
+    TODO
```

Comment:
Please add to the class docstring the following:

1. `self.remap`, `self.inputs`, and `self.outputs` should not be overridden. I was hoping name mangling could allow us to prevent them from being overridden, but unfortunately that won't work (see [example](http://tpcg.io/_7WJQRX)).
2. blackboard values should only be accessed via `blackboard_exists`, `blackboard_get`, and `blackboard_write`, not by accessing `self.blackboard` directly.